### ISSUE TITLE
MSTest project won't compile with WinUI referenced

### ISSUE THREAD

**Issue by dotMorten** [microsoft/microsoft-ui-xaml#4351]
**Describe the bug**
If you create an MSTest project and reference WinUI you get build errors making it impossible to use these APIs in testing.

**Steps to reproduce the bug**
1. Create a new MS Test .NET 5 test project
2. Add WinUI 3 Preview 4
3. Try and compile and observe these errors:
```
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2451,5): warning MSB3842: Project "WinUITests" depends upon SDK "Visual C++ 2015-2019 UWP Desktop Runtime for native apps v14.0" which supports apps targeting "Windows 10.0". To verify whether "Visual C++ 2015-2019 UWP Desktop Runtime for native apps v14.0" is compatible with "Windows 10.0.18362.0", contact the SDK author or see http://go.microsoft.com/fwlink/?LinkID=309181.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2451,5): error MSB3843: Project "WinUITests" targets platform "Windows", but references SDK "Visual C++ 2015-2019 UWP Desktop Runtime for native apps v14.0" which targets platform "UAP".
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2451,5): warning MSB3842: Project "WinUITests" depends upon SDK "Microsoft Universal CRT Debug Runtime v10.0.18362.0" which supports apps targeting "Windows 10.0". To verify whether "Microsoft Universal CRT Debug Runtime v10.0.18362.0" is compatible with "Windows 10.0.18362.0", contact the SDK author or see http://go.microsoft.com/fwlink/?LinkID=309181.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2451,5): error MSB3843: Project "WinUITests" targets platform "Windows", but references SDK "Microsoft Universal CRT Debug Runtime v10.0.18362.0" which targets platform "UAP".
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2451,5): warning MSB3842: Project "WinUITests" depends upon SDK "Visual C++ 2015-2019 Runtime for Universal Windows Platform Apps v14.0" which supports apps targeting "Windows 10.0". To verify whether "Visual C++ 2015-2019 Runtime for Universal Windows Platform Apps v14.0" is compatible with "Windows 10.0.18362.0", contact the SDK author or see http://go.microsoft.com/fwlink/?LinkID=309181.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2451,5): error MSB3843: Project "WinUITests" targets platform "Windows", but references SDK "Visual C++ 2015-2019 Runtime for Universal Windows Platform Apps v14.0" which targets platform "UAP".
```

Full csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0-windows10.0.18362</TargetFramework>
    <TargetPlatformMinVersion>10.0.17134.0</TargetPlatformMinVersion>
    <IsPackable>false</IsPackable>
    <PlatformTarget>x64</PlatformTarget>
    <RootNamespace>WinUITests</RootNamespace>
    <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.WinUI" Version="3.0.0-preview4.210210.4" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.9.1" />
    <PackageReference Include="MSTest.TestAdapter" Version="2.1.2" />
    <PackageReference Include="MSTest.TestFramework" Version="2.1.2" />
    <PackageReference Include="coverlet.collector" Version="3.0.3">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>

</Project>
```

**Expected behavior**
Project compiles fine.

**Screenshots**

![image](https://user-images.githubusercontent.com/1378165/109187468-c9ae6e00-7746-11eb-8ddb-37a5b916a2f9.png)

**Version Info**

NuGet package version:  [Microsoft.WinUI 3.0.0-preview4.210210.4]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        | Yes |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

An alternative approach to reproduce is to create a new WinUI project and add this package to the project:
```
<PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.9.1" />
```

**Comment by StephenLPeters** [comment]
@alwu-msft and @stevenbrix FYI

**Comment by terrycox** [comment]
The same MSB3843 errors occur using NUnit with its own TestAdapter and TestFramework when including Microsoft.NET.Test.Sdk.

**Comment by alwu-msft** [comment]
He @dotMorten, WinUI 3 currently does not support MSTest projects. @JeanRoca is driving investigation from the WinUI end on what this will look like long-term.

**Comment by Scottj1s** [comment]
@alwu-msft - note that this problem is more general than MSTest projects.  It's a regression that was introduced with WinUI 3 Preview 4, which added the VCLibs FWP dependency.  That in turn targets UAP which conflicts with any app that wants to target Windows.  So a simple C++ console app fails to build.

**Comment by alwu-msft** [comment]
@Scottj1s Interesting, I did not realize it was (at least partially) tied to our VCLibs dependency. But it sounds like that's something that the VCLibs owners need to fix, not WinUI?

**Comment by Scottj1s** [comment]
Yes, I think we need help from VCLibs owners on this. $(DesktopCompatible) doesn't cover all scenarios.

**Comment by stevenbrix** [comment]
I filed a bug for this issue here: https://github.com/dotnet/msbuild/issues/6150

There is a workaround in the bug.

I don’t think this has anything to do with our new VCLibs dependency, but it’s due to WebView2 still requiring the old UWP VCLibs. The WebView2 team has moved off UWP VCLibs, and I think we ingested a new WebView2 build recently, so this could be fixed.

To me this seems like an MSBuild issue, a desktop project should be able to reference a “UWP” sdk.

**Comment by Scottj1s** [comment]
The sole purpose of the SDKReferences is to translate them into appx PackageDependencies.  Since those are not needed for unpackaged apps, I've fixed this by enabling the SDKReferences, by default, on TargetPlatformIdentifier == "UAP".  That should unblock previously enabled scenarios like unit tests.

**Comment by dotMorten** [comment]
> I've fixed this by enabling the SDKReferences, by default, on TargetPlatformIdentifier == "UAP

I'm a bit confused by this. The above issue is not for UWP, but .NET 5

**Comment by Scottj1s** [comment]
Sorry - clarified wording

**Comment by dotMorten** [comment]
Thanks Scott. That makes way more sense

**Comment by stevenbrix** [comment]
> unpackaged apps, I've fixed this by enabling the SDKReferences, by default, on TargetPlatformIdentifier == "UAP", and disabling the SDKReferences by default on TargetPlatformIdentifier == "Windows". That should unblock previously enabled scenarios like unit tests.

@Scottj1s packaged desktop apps have `TargetPlatformIdentifer==Windows`, so it sounds like this will break apps once we remove the need for the .wapproj

**Comment by Scottj1s** [comment]
@stevenbrix - yes, the fix as originally implemented did break the new XCG build.  ;-)  I added a condition to support wap-free as well.
